src/Configuration/Programmer/Programmer.cpp

```diff
@@ -183,16 +192,52 @@ void programmer_entry(const CFGCommon_ARG* cmdarg) {
           CFG_POST_ERR("Failed to list devices. Error code: %d", status);
           return;
         }
-        printDeviceList(cable, devices);
+        if (list_device->verbose) {
+          printDeviceList(cable, devices);
+        }
+        if (!devices.empty()) {
+          cmdarg->tclOutput =
+              buildCableDevicesAliasNameWithSpaceSeparatedString(cable,
+                                                                 devices);
+        }
       } else {
-        InitializeHwDb(cableDeviceDb, cableMap, printDeviceList);
+        if (list_device->verbose) {
+          InitializeHwDb(cableDeviceDb, cableMap, printDeviceList);
```

Comment:
this will brake GUI integration since there is if (Gui::GuiInterface()) Gui::GuiInterface()->Devices(cable, deviceList); call inside

Reply:
Wrapped with the new function call `processDeviceList`